src/middleware.ts

```diff
@@ -0,0 +1,10 @@
+import { NextResponse, NextRequest } from 'next/server'
+
+export function middleware(request: NextRequest) {
+  const token = request.nextUrl.searchParams.get('token')
+  const requestHeaders = new Headers(request.headers)
+  if (token) {
+    requestHeaders.set('clientToken', token)
```

Comment:
Naming this `clientToken`, presumably meaning something like "token that http client passes" might conflict with our understanding of user scopes (internalUser / client / company).